### PR TITLE
Possible fix for #1904.

### LINKED ISSUE #1904: Deprecation warnings from numpy
### Steps to reproduce

1. Run pylint over the following test case:

```
"""Test case"""

import numpy as np
value = np.random.seed(1234)
```

### Current behavior
```
/home/bje/source/nemo/myenv/lib/python3.10/site-packages/astroid/raw_building.py:470: FutureWarning: In the future `np.long` will be defined as the corresponding NumPy scalar.  (This may have returned Python scalars in past versions.
  getattr(sys.modules[modname], name)
/home/bje/source/nemo/myenv/lib/python3.10/site-packages/astroid/raw_building.py:470: FutureWarning: In the future `np.long` will be defined as the corresponding NumPy scalar.  (This may have returned Python scalars in past versions.
  getattr(sys.modules[modname], name)
```

### Expected behavior
There should be no future warnings.

### python -c "from astroid import __pkginfo__; print(__pkginfo__.version)" output
2.12.13

### PR DESCRIPTION
**Pull request by bje-**
This may not be a perfect fix for #1904 but it does work. I'll leave this to others more familiar with this code base as to whether this is the best way to fix it.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|     | :sparkles: New feature |
|     | :hammer: Refactoring   |
|     | :scroll: Docs          |

## Description

Closes #1904.

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
Thank you ! We might want to reuse what we did previousely to not duplicate handling, maybe create a decorator and use f-strings. Otherwise LGTM.

File: astroid/raw_building.py [hunk 1]
```diff
@@ -471,7 +477,26 @@ def imported_member(self, node, member, name: str) -> bool:
...
+                with redirect_stderr(io.StringIO()) as stderr, redirect_stdout(
+                    io.StringIO()
+                ) as stdout:
+                    getattr(sys.modules[modname], name)
+                    stderr_value = stderr.getvalue()
+                    if stderr_value:
+                        logger.error(
+                            "Captured stderr while getting %s %s:\n%s",
+                            sys.modules[modname],
+                            name,
```
Comment (cdce8p) [inline astroid/raw_building.py:489]:
```suggestion
                            "Captured stderr while getting %s from %s:\n%s",
                            name,
                            sys.modules[modname],
```
Does this make sense?

File: astroid/raw_building.py [hunk 2]
```diff
@@ -471,7 +477,26 @@ def imported_member(self, node, member, name: str) -> bool:
...
+                            stderr_value,
+                        )
+                    stdout_value = stdout.getvalue()
+                    if stdout_value:
+                        logger.info(
+                            "Captured stdout while importing %s %s:\n%s",
+                            sys.modules[modname],
+                            name,
```
Comment (cdce8p) [inline astroid/raw_building.py:497]: Should be update here too then.

**Review by cdce8p — COMMENTED** [review]
> We might want to reuse what we did previousely to not duplicate handling, maybe create a decorator and use f-strings.

Not sure this makes sense. The implementation is different enough from #1514 where an abstraction would probably complicate it instead of help.

--
A test case like the one from #1514 would be nice.

File: astroid/raw_building.py [hunk 3]
```diff
@@ -471,7 +477,26 @@ def imported_member(self, node, member, name: str) -> bool:
...
+                            sys.modules[modname],
+                            stderr_value,
+                        )
+                    stdout_value = stdout.getvalue()
+                    if stdout_value:
+                        logger.info(
+                            "Captured stdout while importing %s %s:\n%s",
+                            sys.modules[modname],
+                            name,
```
Comment (Pierre-Sassoulas) [inline astroid/raw_building.py:497]:
```suggestion
                            "Captured stout while getting %s from %s:\n%s",
                            name,
                            sys.modules[modname],
```

File: astroid/raw_building.py [hunk 4]
```diff
@@ -471,7 +477,26 @@ def imported_member(self, node, member, name: str) -> bool:
...
+                        logger.error(
+                            "Captured stderr while getting %s from %s:\n%s",
+                            name,
+                            sys.modules[modname],
+                            stderr_value,
+                        )
+                    stdout_value = stdout.getvalue()
+                    if stdout_value:
+                        logger.info(
+                            "Captured stdout while importing %s from %s:\n%s",
```
Comment (Pierre-Sassoulas) [inline astroid/raw_building.py:495]:
```suggestion
                            "Captured stdout while getting %s from %s:\n%s",
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
Tests would be nice, it's possible to reuse the helper in ``test_import_dotted_library``. I tried to do it in order to put this in 2.14.0, but I don't know how to trigger the issue, Mocking ``builtin.getattr`` with ``with unittest.mock.patch(
            "builtin.getattr",
            side_effect=function_with_stdout_and_stderr(expected_out, expected_err),
        ):``did not work.

**Review by Pierre-Sassoulas — APPROVED** [review]
Neat, thank you !

**Pull request merged**